packages/ecs/src/schemas/JSONSchemaTypes.ts

```diff
@@ -0,0 +1,238 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/ir-engine/ir-engine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+export const Kind = Symbol('Kind')
```

Comment:
If this implementation is a copy of derivative of typebox (or any library) we need to explicitly state as such at the top of each file (under our license) as well as include their license in the folder containing the bulk of the copy. We need to respect whatever their license states too.

Reply:
What counts as a derivative? I kept the Kind and static properties, but the rest is unique, Typebox was too inflexible to have typings for Required and NonSerialized fields